Neo4j.Driver/Neo4j.Driver/Internal/ConnectionPool.cs

```diff
@@ -142,10 +148,11 @@ public ConnectionPoolStatus Status
         {
             var connection = await TryExecuteAsync(
                     _logger,
-                    () => AcquireOrTimeoutAsync(mode, database, _connectionAcquisitionTimeout),
+                    () => AcquireOrTimeoutAsync(database, sessionConfig, mode, _connectionAcquisitionTimeout),
```

Comment:
we could probably pack database, mode, sessionConfig and bookmarks into a type. it's quite a logical unit, maybe not one for this PR as it's got enough sprawl as it is.